modules/controlfreec/1.1/controlfreec.smk

```diff
@@ -160,89 +222,128 @@ rule _controlfreec_config:
     params:
         plot = CFG["software"]["FREEC_graph"]
     threads: CFG["threads"]["plot"]
     resources: mem_mb = CFG["mem_mb"]["plot"]
     conda: CFG["conda_envs"]["controlfreec"]
     log: 
-        stdout = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/plot.stdout.log",
-        stderr = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/plot.stderr.log"
+        stdout = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/plot.stdout.log",
+        stderr = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/plot.stderr.log"
+    shell:
+        "cat {params.plot} | R --slave --args `grep \"Output_Ploidy\" {input.info} | cut -f 2` {input.ratios} {input.BAF} > {log.stdout} 2> {log.stderr} "
+
+
+rule _controlfreec_freec2bed:
+    input:
+        ratios = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}.bam_minipileup.pileup.gz_ratio.txt",
+        info = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}.bam_minipileup.pileup.gz_info.txt"
+    output:
+        bed = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}.bed"
+    params:
+        freec2bed = CFG["software"]["freec2bed"]
+    threads: CFG["threads"]["freec2bed"]
+    resources: mem_mb = CFG["mem_mb"]["freec2bed"]
+    conda: CFG["conda_envs"]["controlfreec"]
+    log:
+        stderr = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/freec2bed.stderr.log"
     shell:
-        "cat {params.plot} | R --slave --args `grep \"Output_Ploidy\" {input.info} | cut -f 2` {input.ratios} > {log.stdout} 2> {log.stderr} "
+        "ploidy=$(grep Output_Ploidy {input.info} | cut -f 2); "
+        "perl {params.freec2bed} -f {input.ratios} -p $ploidy > {output.bed} 2> {log.stderr}"
 
 
 # Symlinks the final output files into the module results directory (under '99-outputs/')
 rule _controlfreec_output:
     input:
         plot = str(rules._controlfreec_plot.output.plot),
         log2plot = str(rules._controlfreec_plot.output.log2plot),
-        CNV = str(rules._controlfreec_calc_sig.output.txt)
+        CNV = str(rules._controlfreec_calc_sig.output.txt),
+        bed = str(rules._controlfreec_freec2bed.output.bed),
+        BAF = str(rules._controlfreec_run.output.BAF),
+        BAFgraph = str(rules._controlfreec_plot.output.bafplot),
+        ratio = str(rules._controlfreec_run.output.ratio)
     output:
-        plot = CFG["dirs"]["outputs"] + "{seq_type}--{genome_build}/plots/{sample_id}.bam_ratio.txt.png",
-        log2plot = CFG["dirs"]["outputs"] + "{seq_type}--{genome_build}/log2plots/{sample_id}.bam_ratio.txt.log2.png",
-        CNV = CFG["dirs"]["outputs"] + "{seq_type}--{genome_build}/CNV/{sample_id}.bam_CNVs.p.value.txt"
+        plot = CFG["dirs"]["outputs"] + "{seq_type}--{genome_build}/plots/{tumour_id}--{normal_id}.bam_ratio.txt.png",
```

Comment:
I think bam in all these symlink names can be dropped

Reply:
Good point. Jasper, we've been trying to standardize our output file names. So here, for example, this could be `"{seq_type}--{genome_build}/plots/{tumour_id}--{normal_id}--{pair_status}.controlfree.ratio.png"`

Reply:
Okay I can drop the ".bam" - I haven't integrated pair_status as a wildcard here yet, but I guess it would be a good idea to include it